src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,9 +1,219 @@
-<script setup>
-</script>
 <template>
-    <div class="ma-keywords-generator">
-        <div class="ma-header">
+  <div class="ma-keywords-generator">
+    <div class="card">
+      <div class="container">
+        <div :class="['input-container', { 'centered': isKeywordsGenerated }]">
+          <div class="ma-header">
             <span>Keyword Generator</span>
+          </div>
+          <textarea ref="textarea" v-model="text" placeholder="Enter text here..." rows="6" @input="adjustTextareaHeight"></textarea>
+          <button @click="generateKeywords">Generate Keywords</button>
+        </div>
+        <div v-if="keywords.unigrams.length || keywords.bigrams.length || keywords.trigrams.length" class="keywords-output">
+          <div class="ma-header">
+            <span>Generated Keywords</span>
+          </div>
+          <div class="filter-container">
+            <label for="ngram-filter">Filter by:</label>
+            <select id="ngram-filter" v-model="selectedNGram" @change="filterKeywords">
+              <option value="unigrams">1-grams</option>
+              <option value="bigrams">2-grams</option>
+              <option value="trigrams">3-grams</option>
+            </select>
+          </div>
+          <div>
+            <h4>{{ getNGramLabel() }}</h4>
+            <ul>
+              <li v-for="(item, index) in filteredKeywords" :key="index">{{ item }}</li>
+            </ul>
+          </div>
         </div>
+      </div>
     </div>
-</template>
\ No newline at end of file
+  </div>
+</template>
+
+<script setup>
+import { ref, computed, onMounted } from 'vue';
+
+const text = ref('');
+const keywords = ref({
+  unigrams: [],
+  bigrams: [],
+  trigrams: []
+});
+const selectedNGram = ref('unigrams');
+const isKeywordsGenerated = ref(false);
+const textarea = ref(null);
+
+const generateKeywords = () => {
+  const words = cleanText(text.value).split(/\s+/);
+  keywords.value.unigrams = getNGrams(words, 1);
+  keywords.value.bigrams = getNGrams(words, 2);
+  keywords.value.trigrams = getNGrams(words, 3);
+  isKeywordsGenerated.value = true;
+  adjustTextareaHeight();
+};
+
+const cleanText = (text) => {
+  return text.replace(/[^\w\s]|_/g, '').replace(/\s+/g, ' ').toLowerCase();
+};
+
+const getNGrams = (words, n) => {
+  const nGrams = [];
+  for (let i = 0; i <= words.length - n; i++) {
+    const nGram = words.slice(i, i + n).join(' ');
+    if (!nGrams.includes(nGram)) {
+      nGrams.push(nGram);
+    }
+  }
+  return nGrams;
+};
+
+const getNGramLabel = () => {
+  switch (selectedNGram.value) {
+    case 'unigrams': return '1-grams';
+    case 'bigrams': return '2-grams';
+    case 'trigrams': return '3-grams';
+    default: return '';
+  }
+};
+
+const filteredKeywords = computed(() => {
+  return keywords.value[selectedNGram.value];
+});
+
+const filterKeywords = () => {
+  filteredKeywords.value; // Trigger recompute
+};
+
+const adjustTextareaHeight = () => {
```

Comment:
We don't need a function for textarea height. You can try using `field-sizing: content;`. Or you can just define `rows` directly.
https://developer.mozilla.org/en-US/docs/Web/CSS/field-sizing